service/Core/Handlers/DependencyInjection.cs

```diff
@@ -103,8 +103,7 @@ public static IServiceCollection AddDefaultHandlersAsHostedServices(this IServic
         // Build generic type: HandlerAsAHostedService<THandler>
         Type handlerAsAHostedServiceTHandler = typeof(HandlerAsAHostedService<>).MakeGenericType(tHandler);
 
-        Func<IServiceProvider, IHostedService> implementationFactory =
-            serviceProvider => (IHostedService)ActivatorUtilities.CreateInstance(serviceProvider, handlerAsAHostedServiceTHandler, stepName);
+        IHostedService implementationFactory(IServiceProvider serviceProvider) => (IHostedService)ActivatorUtilities.CreateInstance(serviceProvider, handlerAsAHostedServiceTHandler, stepName);
```

Comment:
what changed here?

Reply:
Created a private function, old code does not compile in my machine so I applied the refactoring to have it compile.

Reply:
@alkampfergit see also this: https://github.com/microsoft/kernel-memory/pull/362#discussion_r1523409039

Reply:
Removed the modified file, after all there is no correlation with this PR it helps reduce the scope.